notifications/NotificationBase.cpp

```diff
@@ -117,6 +117,7 @@ bool CNotificationBase::SendMessageEx(
 		fText = CURLEncode::URLEncode(fText);
 	}
 
+	boost::mutex::scoped_lock SendMessageEx(SendMessageExMutex);
```

Comment:
To clarify, I mean calls to CNotificationHelper::SendMessageEx. Locking this one up would defeat the purpose of using threads (well, not entirely, it would allow to run the notifiers simultaneously), so best alternative is locking up the spawned threads.